### PR TITLE
Partition Step Run and Remove Prisma

### PR DESCRIPTION
**Pull request by reillyse**
Adding partitions to step_run in a migration

### CODE REVIEW

File: Taskfile.yaml [hunk 1]
```diff
@@ -52,18 +52,22 @@ tasks:
   stop-db:
     cmds:
       - docker compose down
+  recreate-db-from-scratch:
```
Comment (grutt) [inline Taskfile.yaml:55]: nice

File: pkg/repository/prisma/dbsqlc/sqlc.yaml [hunk 2]
```diff
@@ -25,7 +25,7 @@ sql:
       - queue.sql
       - lease.sql
     schema:
-      - schema.sql
+      - ../../../../sql/schema/schema.sql
```
Comment (grutt) [inline pkg/repository/prisma/dbsqlc/sqlc.yaml:28]: what is happening here now... why did this ref change?

File: pkg/repository/prisma/dbsqlc/step_runs.sql [hunk 3]
```diff
@@ -664,6 +664,7 @@ WITH step_runs_on_inactive_workers AS (
     WHERE
         w."tenantId" = @tenantId::uuid
         AND w."lastHeartbeatAt" < NOW() - INTERVAL '30 seconds'
+        AND sr."status" in ('PENDING','PENDING_ASSIGNMENT','ASSIGNED','RUNNING','CANCELLING' )
```
Comment (grutt) [inline pkg/repository/prisma/dbsqlc/step_runs.sql]: is this now required due to partitioning?
why is this change here?

File: sql/migrations/20241028211833_v0.51.0.sql
Comment (grutt) [inline sql/migrations/20241028211833_v0.51.0.sql:1]: i think version num is off now?

**Review by grutt — CHANGES_REQUESTED** [review]
looks good pending the version num changes

File: pkg/repository/prisma/dbsqlc/sqlc.yaml (hunk 2 above)
Comment (reillyse) [reply pkg/repository/prisma/dbsqlc/sqlc.yaml:28]: iirc we were processing the schema and moving it to this location, now we are just using the schema and skipping the prisma piece

File: pkg/repository/prisma/dbsqlc/step_runs.sql (hunk 3 above)
Comment (reillyse) [reply pkg/repository/prisma/dbsqlc/step_runs.sql]: Probably more important before partitioning actually  - it's probably redundant now because we partition on that column

File: sql/migrations/20241028211833_v0.51.0.sql
Comment (reillyse) [reply sql/migrations/20241028211833_v0.51.0.sql:1]: ok, I've moved them around. Do I need to recreate the timestamp?

**Review by grutt — APPROVED** [review]
